docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,274 @@
+
+## Orca Estimator
+After initializing the OrcaContext, we could simply create an Estimatornow, and the Estimator will replicate the model on each node in the cluster, feed the data partition on each node to the local model replica, and synchronize the model parameters using various backend technologies.
+
+Let's get started!
+# Prepare Environment
+Before submitting the BigDL program to Yarn Cluster, we need to setup the environment as the following steps:
+* Setup JAVA Environment
+* Setup Spark Environment
+* Setup Hadoop Environment
+* Install All Needed Python Libraries 
+* For CDH Users
+
+## Setup JAVA Environment
+We need to download and install JDK in the environment, and properly set the environment variable JAVA_HOME, which is required by Spark. JDK8 is highly recommended.
+
+```bash
+# For Ubuntu
+sudo apt-get install openjdk-8-jre
+export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/
+
+# For CentOS
+su -c "yum install java-1.8.0-openjdk"
+export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.282.b08-1.el7_9.x86_64/jre
+
+export PATH=$PATH:$JAVA_HOME/bin
+java -version  # Verify the version of JDK.
+```
+
+## Setup Spark Environment
+We need to set environment variables `${SPARK_HOME}` as follows:
+```bash
+export SPARK_HOME=the folder path where you extract the Spark package
+```
+
+## Setup Hadoop Environment
+Check the Hadoop setup and configurations of our cluster. Make sure we correctly set the environment variable HADOOP_CONF_DIR, which is needed to initialize Spark on YARN:
+
+```bash
+export HADOOP_CONF_DIR=the directory of the hadoop and yarn configurations
+```
+
+## Install Needed Python Libraries
+We need first to use conda to prepare the Python environment on the local machine where we submit our application. Create a conda environment, install BigDL and all the needed Python libraries in the created conda environment:
+``` bash
+conda create -n bigdl python=3.7  # "bigdl" is conda environment name, you can use any name you like.
+conda activate bigdl
+# Use conda or pip to install all the needed Python dependencies in the created conda environment.
+pip install bigdl
+pip install conda-pack # Use conda-pack to pack the conda environment to an archive
+```
+### Options
+* When you are running a program on Ray backend, please install Ray as below:
+```bash
+pip install ray[default]
+```
+* When you are running a TensorFlow2 model, please install TensorFlow as below:
+```bash
+pip install tensorflow==2.6.0 keras==2.6.0 # When you are running a tensorflow model.
+```
+
+## For CDH Users
+If your CDH cluster has already installed Spark, the CDH’s Spark might be conflict with the pyspark installed by pip required by BigDL. Thus before running BigDL applications, you should unset all the Spark related environment variables. You can use `env | grep SPARK` to find all the existing Spark environment variables.
```

Comment:
Is there a better way to handle this? This would require the user to change the environment variables on each node in the cluster. Can we use `spark.yarn.appMasterEnv` or `spark.executorEnv`? Or make sure the installed Spark version matches the CDH version?

Reply:
that's mainly asks users to unset  `spark_home` on driver node, but installing a match version would be better.

Reply:
Make it specific what is needed to run on CDH.